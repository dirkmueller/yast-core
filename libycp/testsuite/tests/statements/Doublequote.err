Parsed:
----------------------------------------------------------------------
{ return (1 + 2); }
----------------------------------------------------------------------
tests/statements/Doublequote.ycp:18:Undeclared identifier 'name'
[Parser] tests/statements/Doublequote.ycp:18 Undeclared identifier 'name'
Parsed:
----------------------------------------------------------------------
YError
----------------------------------------------------------------------
[Interpreter] tests/statements/Doublequote.ycp:18 Parser error
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/statements/Doublequote.ycp"
    return (3 + 4);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/statements/Doublequote.ycp"
    return 7;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // block <integer> foo1
    // filename: "tests/statements/Doublequote.ycp"
    block <integer> foo1 = nil;
    foo1 = { return {
    return 42;
}
; };
    return eval (foo1);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // block <integer> foo1
    // integer foo2
    // filename: "tests/statements/Doublequote.ycp"
    block <integer> foo1 = nil;
    integer foo2 = 55;
    foo1 = { return {
    return (foo2 + 1);
}
; };
    foo2 = 41;
    return eval (foo1);
}
----------------------------------------------------------------------
