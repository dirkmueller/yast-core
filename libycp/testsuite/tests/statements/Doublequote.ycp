
# ---------------------------------------------------------
#
#  Filename:    Builtin_doubleqoute.ycp
#
#  Purpose:     test cases for YCP doubleqoute
#
#  Creator:     jd@suse.de
#
#  Maintainer:  jd@suse.de
#
# ---------------------------------------------------------

// The double backquotes `` are called "deep quote". They are used
// to prevent the enclosed expression from evaluation.

( ``( 1+2 ))
( ``(name) )
( ``{ return 3 + 4; } )
( { return 7; } )

( {
	// this code tests block assigning (not in variable definition)
	block <integer> foo1 = nil;
	
	foo1 = ``{ return 42; };
	
	return eval(foo1);
  })
  
( {
	// this code tests block assigning (not in variable definition)
	// also, the code must not be evaluated
	block <integer> foo1 = nil;
	
	integer foo2 = 55;
	
	foo1 = ``{ return foo2+1; };
	
	foo2 = 41;
	
	return eval(foo1);
  })