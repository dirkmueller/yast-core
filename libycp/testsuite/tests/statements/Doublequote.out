((1 + 2))
(nil)
(7)
(7)
(42)
(42)
